omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_4/FrontendJsonConfigController2_4.java

```diff
@@ -0,0 +1,155 @@
+
+    public static final String DEFAULT_FRONTEND_DIRECTORY = "frontend";
+    public static final String GP_LOCAL_DIRECTORY = "spa.local.directory";
+    private static final String JSON_CONFIG_FILE_NAME = "config.json";
+
+    private Log log = LogFactory.getLog(this.getClass());
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void saveFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String basicAuth = request.getHeader("Authorization");
+        boolean isAuthenticated = Context.isAuthenticated();
+        if (!isAuthenticated && basicAuth == null) {
+            response.sendError(HttpServletResponse.SC_FORBIDDEN, "No Authorisation Provided");
+            return;
+        }
+        if (!isAuthenticated && isValidAuthFormat(response, basicAuth)) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid credentials provided");
+            return;
+        }
+        User user = Context.getAuthenticatedUser();
+        if (user == null || !user.isSuperUser()) {
+            log.error("Authorization error while creating a config.json file");
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+        saveJsonConfigFile(request, response);
+    }
+
+    private void saveJsonConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        try {
+            BufferedReader reader = request.getReader();
+            StringBuilder stringBuilder = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                stringBuilder.append(line);
+            }
+            String requestBody = stringBuilder.toString();
+
+            new ObjectMapper().readTree(requestBody); // verify that is in a valid JSON format
+
+            InputStream inputStream = new ByteArrayInputStream(requestBody.getBytes(StandardCharsets.UTF_8));
+            OutputStream outStream = Files.newOutputStream(jsonConfigFile.toPath());
+            OpenmrsUtil.copyFile(inputStream, outStream);
+
+            if (jsonConfigFile.exists()) {
+                log.debug("file: '{" + jsonConfigFile.getAbsolutePath() + "}' written successfully");
+                response.setStatus(HttpServletResponse.SC_OK);
+            }
+        } catch (JsonProcessingException e) {
+            log.error("Invalid JSON format", e);
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    private boolean isValidAuthFormat(HttpServletResponse response, String basicAuth) {
+        if (basicAuth.startsWith("Basic")) {
+            try {
+                // remove the leading "Basic "
+                basicAuth = basicAuth.substring(6);
+                if (StringUtils.isBlank(basicAuth)) {
+                    return true;
```

Comment:
Is it a valid auth format when this is blank?

Reply:
i implemented the logic vice versa, i will rename the method to  `isInvalidAuthFormat `